packages/core/src/local-workspace/remote_map/remote_map.ts

```diff
@@ -443,36 +427,11 @@ ReadIterator => {
 }
 
 export const createRemoteMapCreator = (location: string,
-  persistentDefaultValue = false,
-  cacheSize = 5000):
+  persistentDefaultValue = false):
 remoteMap.RemoteMapCreator => {
-  const counterInc = (counter: CounterType): void => {
-    counters[location][counter] += 1
-  }
-  if (location in counters) {
-    log.debug('Expected one RemoteMapCreator per location, but looks like a more than one is being created')
-  } else {
-    counters[location] = Object.fromEntries(
-      COUNTER_TYPES.map(counterName => [counterName, 0])
-    ) as Record<CounterType, number>
-  }
-
-  // Note: once we set a non-zero cache size,
-  // we won't change the cache size even if we give different value
-  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
-  let locationCache: LRU<string, any>
-  if (locationCaches.has(location)) {
-    counterInc('LocationCacheReuse')
-    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
-    locationCache = locationCaches.get(location) as LRU<string, any>
-  } else {
-    counterInc('LocationCacheCreated')
-    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
-    locationCache = new LRU<string, any>({ max: cacheSize })
-    if (cacheSize > 0) {
-      locationCaches.set(location, locationCache)
-    }
-  }
+  const locationResources = remoteMapLocations.get(location)
+  const statCounters = locationResources.counters
+  const locationCache = locationResources.cache
```

Comment:
```suggestion
  const { counters: statCounters, cache: locationCache } = remoteMapLocations.get(location)
```
just to avoid having that extra name...